src/libcore/iter.rs

```diff
@@ -928,7 +885,7 @@ pub trait IteratorExt: Iterator + Sized {
     /// `std::usize::MAX` elements of the original iterator.
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
-    fn rev(self) -> Rev<Self> {
+    fn rev(self) -> Rev<Self> where Self: Sized {
```

Comment:
Can you make this `where Self: Sized + DoubleEndedIterator` in order to fix #23587?